File: ios/SwiftKotlination/SwiftKotlinationUITests/Sources/Extensions/Parameters/Parameters+Query.swift
extension Parameters {

    var query: String? {
        switch self {
        case .url(let url):
            guard !url.isEmpty else {
                return nil
            }

            return url
                .map { key, value in
                    guard let value = value else {
                        return key
                    }
                    return "\(key)=\(value)"
                }.joined(separator: "&")
        default:
            return nil
        }
    }
}

File: android/SwiftKotlination/app/src/androidTest/kotlin/fr/jhandguy/swiftkotlination/extensions/Parameters+Query.kt
package fr.jhandguy.swiftkotlination.extensions

import fr.jhandguy.network.model.network.Parameters

fun Parameters.query(): String? =
    when (this) {
        is Parameters.Url -> {
            url.joinToString(separator = "&") { query ->
                query.second?.let {
                    "${query.first}=${query.second}"
                } ?: query.first
            }
        }
        else -> null
    }
